### PR TITLE
Certificate authentication

### PR DESCRIPTION
**Pull request by mariusmihaic**
Generate certificate (.crt + .pem) for gRPC TLS connection between server and client.
--

**Certificate Generation**: To enable secure communication, generate a certificate pair containing a .crt (certificate) and a .pem (private key) for both the server and the sovereign nodes (clients). This will facilitate the encryption and authentication required for the gRPC TLS connection.

**Authentication of Clients**: The server, acting as the hot wallet binary, should authenticate and validate the sovereign nodes (clients) attempting to connect. Only trusted clients with the matching certificate will be granted access to interact with the hot wallet binary.

**Ensuring Secure Transactions:** Utilize the certificate-based authentication mechanism to ensure that only authorized sovereign nodes can access the hot wallet binary. This step is crucial in maintaining the integrity and security of transactions being sent from the sovereign shards to the main chain.

**Ongoing Security Measures**: Regularly review and update the certificate mechanism to maintain security. This includes renewal of certificates, implementing security best practices, and promptly revoking access for compromised or unauthorized clients.

### CODE REVIEW

File: server/cmd/server/main.go [hunk 1]
```diff
@@ -9,12 +9,15 @@ import (
 	"syscall"
 	"time"
 
+	"google.golang.org/grpc/credentials"
```
Comment (sasurobert) [inline server/cmd/server/main.go:12]: goimports

File: client/cmd/client/.env [hunk 2]
```diff
@@ -2,3 +2,9 @@
 GRPC_HOST="localhost"
 # GRPC server port
 GRPC_PORT="8085"
+# Client certificate for tls secured connection with server.
```
Comment (raduchis) [inline client/cmd/client/.env:5]: why .env file and not config.toml?

File: client/cmd/client/main.go [hunk 3]
```diff
@@ -136,13 +139,22 @@ func loadConfig() (*config.ClientConfig, error) {
...
+	certFile := os.Getenv(envCertFile)
+	certPkFile := os.Getenv(envCertPkFile)
 
 	log.Info("loaded config", "grpc host", grpcHost)
 	log.Info("loaded config", "grpc port", grpcPort)
 
+	log.Info("loaded config", "certificate file", certFile)
+	log.Info("loaded config", "certificate pk", certPkFile)
+
 	return &config.ClientConfig{
```
Comment (raduchis) [inline client/cmd/client/main.go:151]: the env variables could be loaded directly from a config.toml file

File: server/cmd/server/.env [hunk 4]
```diff
@@ -12,3 +12,9 @@ MULTIVERSX_PROXY="https://testnet-gateway.multiversx.com"
 BRIDGE_SC_ADDRESS="erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx"
 # Max retrials to wait for account nonce update after sending bridge txs
 MAX_RETRIALS_WAIT_NONCE=60
+# Server certificate for tls secured connection with clients.
```
Comment (raduchis) [inline server/cmd/server/.env:15]: also here why not a config.toml file?

File: server/cmd/server/main.go (hunk 1 above)
Comment (mariusmihaic) [reply server/cmd/server/main.go:12]: Indeed looks strange, but It is already formatted, that's how it looks.

File: client/cmd/client/.env (hunk 2 above)
Comment (mariusmihaic) [reply client/cmd/client/.env:5]: Several reasons:
- for small projects or simple setups, using a single .env file can be straightforward and easy to manage.
- docker Compatibility: .env files are commonly used in Docker containers, and they integrate well with container orchestration tools like Docker Compose and Kubernetes. They allow for easy configuration changes without modifying the application code, which is very useful.

There is already a task created to make it runnable with either .env/configs/flags.

File: client/cmd/client/main.go (hunk 3 above)
Comment (mariusmihaic) [reply client/cmd/client/main.go:151]: Several reasons:
- for small projects or simple setups, using a single .env file can be straightforward and easy to manage.
- docker Compatibility: .env files are commonly used in Docker containers, and they integrate well with container orchestration tools like Docker Compose and Kubernetes. They allow for easy configuration changes without modifying the application code, which is very useful.

There is already a task created to make it runnable with either .env/configs/flags.

File: server/cmd/server/.env (hunk 4 above)
Comment (mariusmihaic) [reply server/cmd/server/.env:15]: Several reasons:
- for small projects or simple setups, using a single .env file can be straightforward and easy to manage.
- docker Compatibility: .env files are commonly used in Docker containers, and they integrate well with container orchestration tools like Docker Compose and Kubernetes. They allow for easy configuration changes without modifying the application code, which is very useful.

There is already a task created to make it runnable with either .env/configs/flags.

**Review by raduchis — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**